source/views/settings/icon.js

```diff
@@ -0,0 +1,95 @@
+// @flow
+import React from 'react'
+import {ScrollView, Image, StyleSheet, Text} from 'react-native'
+import Icons from 'react-native-alternate-icons'
+import {Section, Cell} from 'react-native-tableview-simple'
+import {Column} from '../components/layout'
+import includes from 'lodash/includes'
+
+const styles = StyleSheet.create({
+  content: {
+    flex: 1,
+    paddingVertical: 10,
+  },
+  title: {
+    fontSize: 16,
+  },
+  icon: {
+    width: 16,
+    height: 16,
+    borderRadius: 5,
+    borderWidth: 0.5,
+  },
+})
+
+const icons = [
+  {
+    type: 'default',
+    src: require('../../../images/about/IconTrans.png'),
+    title: 'Old Main',
+  },
+  {
+    type: 'icon_type_windmill',
+    src: require('../../../ios/AllAboutOlaf/windmill.png'),
+    title: 'Wind Turbine (Big Ole)',
+  },
+]
+
+export default class IconSettingsView extends React.PureComponent {
+  static navigationOptions = {
+    title: 'App Icon',
+  }
+
+  state: {
+    iconType: string,
+  } = {
+    iconType: '',
+  }
+
+  componentWillMount() {
+    this.getIcon()
+  }
+
+  setIcon(iconType: string) {
+    iconType == 'default' ? Icons.reset() : Icons.setIconName(iconType)
```

Comment:
===

Reply:
Actually, I think I'd like this to go into a full-fledged if-else statement – something about executing side-effects in a ternary feels odd

Reply:
To follow up on that, this logic has a race condition. Because the library uses callbacks, not promises, the only way to guarantee that come code will be executed after other code is to run it _in_ the callback. You'll either need to wrap the library in promises (use pify, works well) or just call this.getIcon in the callback of each logic branch here.

Reply:
Wait, do these not accept callbacks?

Reply:
Despite using completion handlers (thus callbacks) in the objc code, these two functions do not take a callback. 

 We'll need to PR that upstream before I'm comfortable accepting this, sadly.

Reply:
https://github.com/kamsteegsoftware/react-native-alternate-icons/blob/122be1d1296efb87dcaf0075467b879a1be25c0d/ReactNativeAlternateIcons.m#L18

Reply:
Okay. We’ll PR for callbacks.

Reply:
I've changed `setIcon` to an if/else block to avoid executing side-effects in a ternary. Although this block collapses, we'll still need to PR upstream to get the objc code to take a callback.